### ISSUE TITLE
Activity View Crashes with Skills Having Parent Skills

### ISSUE THREAD

**Issue by mheggeseth** [HTBox/allReady#487]
The activity view (`Views/Activity/Activity.cshtml`) crashes if any of the activity's required skills have a parent skill or if any of the user's associated skills have a parent skill.

@stevejgordon This appears to be the result of adding the `Children` property to the `Skill` class. It causes a circular reference exception when trying to JSON serialize. Is the property strictly needed or can you get `SkillDeleteQueryHandlerAsync` to work without it?

**Comment by stevejgordon** [comment]
@mheggeseth Sorry about that! I've just submitted a PR to address this. I've done as you suggested and removed the need for the property. The only reason I'd added it was to avoid the need to query the DB with two calls when we get the skill and then the list of children. There is a minor efficiency gain to be able to include it all in one query.

One other solution I tested which we could consider to leave the property in place but to add the [JsonIgnore] attribute. I did try that and it seems to work. So if we like that option I could amend this PR.
